kofserver/scoreboard.py

```diff
@@ -97,6 +121,71 @@ def LogPlayerAction(gameName, playerName, actionType, actionContent):
     # QueryScore("game1", "Alice") should results in:
     # [ { "playerName": "Alice", "portUptime": 0, "portScore": 0,
     #     "pidUptime": 60, "pidScore": 300, "totalScore": 300 } ]
-    def QueryScore(gameName, playerName):
-        # TODO
-        return []
+    def QueryScore(self, gameName, playerName):
+        sql = "SELECT player_name, SUM(port_uptime), SUM(port_score_per_sec), SUM(pid_uptime), SUM(pid_score_per_sec) FROM score %s GROUP BY game_name, player_name;"
+        whereSQL = ""
+        if gameName != "":
+            whereSQL += "WHERE game_name='%s'" % (gameName)
```

Comment:
SQL injection is possible here, use "?" instead.
https://docs.python.org/2/library/sqlite3.html
"Instead, use the DB-API’s parameter substitution. Put ? as a placeholder wherever you want to use a value, and then provide a tuple of values as the second argument to the cursor’s execute() method."

Reply:
Done